packages/lib/server/checkBookingLimits.ts

```diff
@@ -40,55 +42,45 @@ export async function checkBookingLimit({
 }: {
   eventStartDate: Date;
   eventId: number;
-  key: string;
-  limitingNumber: number;
+  key: keyof IntervalLimit;
+  limitingNumber: number | undefined;
   returnBusyTimes?: boolean;
 }) {
   {
-    const limitKey = key as keyof IntervalLimit;
-    // Take PER_DAY and turn it into day and PER_WEEK into week etc.
-    const filter = limitKey.split("_")[1].toLocaleLowerCase() as "day" | "week" | "month" | "year"; // Have to cast here
-    const startDate = dayjs(eventStartDate).startOf(filter).toDate();
-    // this is parsed above with parseBookingLimit so we know it's safe.
+    if (!limitingNumber) return;
 
-    const endDate = dayjs(startDate).endOf(filter).toDate();
+    const unit = intervalLimitKeyToUnit(key);
+
+    const startDate = dayjs(eventStartDate).startOf(unit).toDate();
+    const endDate = dayjs(eventStartDate).endOf(unit).toDate();
 
-    // This allows us to easily add it within dayjs
     const bookingsInPeriod = await prisma.booking.count({
       where: {
-        status: "ACCEPTED",
-        eventType: {
-          AND: [
-            {
-              id: eventId,
-              bookings: {
-                every: {
-                  startTime: {
-                    gte: startDate,
-                  },
-                  endTime: {
-                    lte: endDate,
-                  },
```

Comment:
This is a strange recursion that would not count bookings if not ALL bookings for that event were in the same period